src/Propulsion.cpp

```diff
@@ -36,65 +37,143 @@ void Propulsion::LoadFromJson(const Json::Value &jsonObj, Space *space)
 	SetThrusterState( temp_vector );
 	m_thrusterFuel = StrToDouble(jsonObj["thruster_fuel"].asString());
 	m_reserveFuel = StrToDouble(jsonObj["reserve_fuel"].asString());
+
+	//m_gThrusters->LoadFromJson(jsonObj);
 	// !!! This is commented to avoid savegame bumps:
 	//m_fuelTankMass = jsonObj["tank_mass"].asInt();
 };
 
+void Propulsion::SaveNacelleToJson(Json::Value &jsonObj)
+{
+	printf("Propulsion nacelle save\n");
+	Json::Value nacellePropArray(Json::arrayValue);
+	for (unsigned int i=0; i<m_vectThVector.size(); i++) {
+		Json::Value nacellePropArrayEl(Json::objectValue); // Create JSON object to contain nacelle.
+		nacellePropArrayEl["nacelle_power"]=m_vectThVector[i].powLevel;
+		nacellePropArray.append(nacellePropArrayEl);
+	}
+	jsonObj["nacelle_data"] = nacellePropArray;
+}
+
+void Propulsion::LoadNacelleFromJson(const Json::Value &jsonObj)
+{
+	printf("Propulsion nacelle load\n");
+	Json::Value nacellePropArray = jsonObj["nacelle_data"];
+	if (!nacellePropArray.isArray()) throw SavedGameCorruptException();
+	assert(nacellePropArray.size()==m_vectThVector.size());
+	for (unsigned int i=0; i<m_vectThVector.size(); i++) {
+		Json::Value nacellePropArrayEl = nacellePropArray[i];
+		if (!nacellePropArrayEl.isMember("nacelle_power")) throw SavedGameCorruptException();
+
+		m_vectThVector[i].powLevel = nacellePropArrayEl["nacelle_power"].asFloat();
+	}
+}
+
 Propulsion::Propulsion()
 {
 	m_power_mul = 1.0;
 	m_fuelTankMass = 1;
 	for ( int i=0; i< Thruster::THRUSTER_MAX; i++) m_linThrust[i]=0.0;
 	m_angThrust = 0.0;
-	m_effectiveExhaustVelocity = 100000.0;
+	m_effectiveExhaustVelocity = 1000.0; // This should have an impact on game...
```

Comment:
m_effectiveExhaustVelocity need to be initialized to a value.

But when you play you will overwrite this default value with data loaded from
json files: if this not happen then something is wrong :/

Here I would provide a feedback to players/modders/ecc: if this value is
used when they play, then your tank will empty very very fast.

...Probably a better comment should be useful...